dEdX/src/dEdxAnalyser.cc

```diff
@@ -172,37 +202,44 @@ void dEdxAnalyser::init() {
   theDet.field().magneticField( { 0., 0., 0. }  , bfieldV  ) ;
   _bField = bfieldV[2]/dd4hep::tesla ;
 
-  streamlog_out(MESSAGE) << "geometry prepared" << std::endl;
+  streamlog_out(DEBUG) << "geometry prepared" << std::endl;
 
   // prepare bins and string content for histograms
   //const int nBinsX=100, nBinsY=300;
   //double minBinX=-1, maxBinX=2, minBinX2=-3;  // log10(min/max momentum / GeV)
   //double minBinY= 0, maxBinY=1e-6;  // for dE/dx values in GeV/mm
   //double histbinsY[nBinsY+1], histbinsX2[nBinsX+1];  //histbinsX[nBinsX+1]
-  double* histbinsX = new double[_nBinsX+1];
-  double* histbinsX2= new double[_nBinsX+1];
-  double* histbinsY = new double[_nBinsY+1];
+  double* histbinsX  = new double[_nBinsX+1];
+  double* histbinsX2 = new double[_nBinsX+1];
+  double* histbinsX3 = new double[_nBinsX+1];
+  double* histbinsY  = new double[_nBinsY+1];
+  double* histbinsY2 = new double[_nBinsY+1];
```

Comment:
Some of these are leaked.
```suggestion
  double* histbinsX  = std::vector<double>(_nBinsX + 1);
  double* histbinsX2 = std::vector<double>(_nBinsX + 1);
  double* histbinsX3 = std::vector<double>(_nBinsX + 1);
  double* histbinsY  = std::vector<double>(_nBinsY + 1);
  double* histbinsY2 = std::vector<double>(_nBinsY + 1);
```

And then use  via `_nBinsX.data()` for constructing histograms.